src/main/java/com/puppycrawl/tools/checkstyle/checks/imports/ImportOrderCheck.java

```diff
@@ -443,12 +449,26 @@ else if (groupIdx > lastGroup) {
         else {
             log(line, MSG_ORDERING, name);
         }
-
+        checkRedundantSeparation(groupIdx, isStatic, line, name);
         lastGroup = groupIdx;
         lastImport = name;
     }
 
     /**
+     * Checks whether group of imports have redundant separation
+     * @param groupIdx group number.
+     * @param isStatic whether the token is static or not.
+     * @param line the line of the current import.
+     * @param name the name of the current import.
+     */
+    private void checkRedundantSeparation(int groupIdx, boolean isStatic, int line, String name) {
+        if (!beforeFirstImport && separated && groupIdx == lastGroup
+                && isStatic == lastImportStatic && line - lastImportLine > 1) {
+            log(line, MSG_REDUNDANT_SEPARATION, name);
```

Comment:
please try to return boolean from this variable and do "log" in visitToken to let easily see when log of violation happen.